File: algorithms/algorithms-cplusplus/leetcode/Question_0080_Remove_Duplicates_from_Sorted_Array_II.cc
#include <vector>

using namespace std;

class Solution {
 public:
  int removeDuplicates(vector<int>& nums) {
    if (nums.empty()) {
      return 0;
    }
    size_t index = 0;
    int count = 0;
    for (size_t i = 0; i < nums.size(); i++) {
      if (i > 0 && nums[i - 1] == nums[i]) {
        count++;
        if (count >= 2) {
          continue;
        }
      } else {
        count = 0;
      }
      nums[index++] = nums[i];
    }
    return index;
  }
};

File: algorithms/algorithms-python/leetcode/Question_0080_Remove_Duplicates_from_Sorted_Array_II.py
#!/usr/bin/python
# coding: utf-8

class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return 0
        index, count = 0, 0
        for i in range(len(nums)):
            if i != 0 and nums[i - 1] == nums[i]:
                count += 1
                if count >= 2:
                    continue
            else:
                count = 0
            nums[index] = nums[i]
            index += 1
        return index
